### ISSUE TITLE
make 'arrow-body-style' optional

### ISSUE THREAD

**Issue by lfender6445** [rentpath/eslint-config-rentpath#24]
I vote we make this optional.

Its a pain, especially when debugging, to convert your function structure, when all you really wanted was to console.log an arg.

**Comment by colinrymer** [comment]
Can you share some example code?

**Comment by lfender6445** [comment]
sure its the difference between this:

```

**Comment by c0** [comment]
I agree that it is painful when debugging and we've seen some bugs come out when curlies are used without a return (b/c it's assumed it will return).

I would like to propose another option:

* `()` is required for single-line arrow functions. 
* Multi-line arrow functions require `{}` and `return`

### Examples

**Allowed**
```js
const mapStateToProps = state => ({ cities: state.content.get('cities') })

const mapStateToProps = state => {
  return { cities: state.content.get('cities') }
}
```

**Disallowed**
```js
const mapStateToProps = state => { return { cities: state.content.get('cities') } }

const mapStateToProps = state => (
  { cities: state.content.get('cities') }
)
```

**Comment by warrenv** [comment]
so the proposal is to disallow implicit returns?

**Comment by c0** [comment]
@warrenv Luke is proposing making implicit returns optional.

I'm proposing that we disallow implicit returns on multi-lines.

**Comment by warrenv** [comment]
@lfender6445 it makes for much fewer vim command s If you use the vim-surround plugin.
https://github.com/tpope/vim-surround

**Comment by wedgex** [comment]
I'm fine with making it optional, I don't want to disallow it on multilines because it will make writing stateless components alot messier ie it would disallow writing components like this:

```jsx
  export default ({ thing }) => (
    <div>
       <span>{ thing }</span>
    </div>
  )
```

**Comment by lfender6445** [comment]
@warrenv i thought about vim bindings, but this applies to any and every editor really

**Comment by lfender6445** [comment]
so how about single line required, multiline optional?

**Comment by c0** [comment]
@wedgex Very good point about JSX. I don't want to lose that either.
